diff --git a/lib/utils.sh b/lib/utils.sh
index 0fcaea0..ec138d0 100644
--- a/lib/utils.sh
+++ b/lib/utils.sh
@@ -153,8 +153,10 @@
     
     # 展开 spec 文件中的宏定义（目的：方便获取子包名）
     rpmspec -P $SPEC_FILE > $NEW_SPEC_FILE
-
     local main_pkg=("$(grep '^Name:' "$NEW_SPEC_FILE" | sed 's/^Name:\s*//')")
+    
+    sed -n '/^%install/,/^%changelog/p' $NEW_SPEC_FILE > tmp.spec
+    mv -f tmp.spec $NEW_SPEC_FILE
 
     # spec 文件中 %files 行信息（用于查询含有man手册的包）
     local files_messages=()
