diff -Nuar cqpkg_manager-1.0.1/usr/bin/clone cqpkg_manager-1.0.1-diff/usr/bin/clone
--- cqpkg_manager-1.0.1/usr/bin/clone	2025-06-24 14:41:27.340311333 +0800
+++ cqpkg_manager-1.0.1-diff/usr/bin/clone	2025-06-24 14:42:07.084728544 +0800
@@ -275,10 +275,10 @@
     fi
     
     # 获取仓库类型
-    read -p "请选择仓库类型（a=欧拉, b=龙蜥, 其他=CQ内部仓库）: " REPO_TYPE
+    read -e -p "请选择仓库类型（a=欧拉, b=龙蜥, 其他=CQ内部仓库）: " REPO_TYPE
     
     # 获取分支名
-    read -p "请输入克隆分支（默认 $DEFAULT_BRANCH）: " input_branch
+    read -e -p "请输入克隆分支（默认 $DEFAULT_BRANCH）: " input_branch
     if [[ -n "$input_branch" ]]; then
         CLONE_BRANCH="$input_branch"
     fi
@@ -286,7 +286,7 @@
     PKG_PATH="$(pwd)/$PKG_NAME"
     
     # 是否创建rpmbuild目录
-    read -p "是否将代码放入$HOME/rpmbuild目录（输入y放入此目录，默认放到 $PKG_PATH 目录）: " CREATE_RPMBUILD_DIR
+    read -e -p "是否将代码放入$HOME/rpmbuild目录（输入y放入此目录，默认放到 $PKG_PATH 目录）: " CREATE_RPMBUILD_DIR
     
     # 设置工作路径
     if [[ "$CREATE_RPMBUILD_DIR" == "y" ]]; then
diff -Nuar cqpkg_manager-1.0.1/usr/bin/cqpatch cqpkg_manager-1.0.1-diff/usr/bin/cqpatch
--- cqpkg_manager-1.0.1/usr/bin/cqpatch	2025-06-24 14:41:27.340311333 +0800
+++ cqpkg_manager-1.0.1-diff/usr/bin/cqpatch	2025-06-24 14:43:02.788469332 +0800
@@ -102,7 +102,7 @@
     log_info "修改完成后输入补丁文件名继续..."
     
     local patch_name
-    read -p "请输入补丁文件名（例：cqos-func-add-chinese-man-page.patch）: " patch_name
+    read -e -p "请输入补丁文件名（例：cqos-func-add-chinese-man-page.patch）: " patch_name
     if [ -z "$patch_name" ]; then
         log_error "补丁文件名不能为空"
         exit 1
@@ -205,7 +205,7 @@
     local user_email=$(git config --global user.email)
     
     if [ -z "$user_name" ]; then
-        read -p "请设置用户名: " user_name
+        read -e -p "请设置用户名: " user_name
         if [ -z "$user_name" ]; then
             log_error "用户名不能为空"
             exit 1
@@ -214,7 +214,7 @@
     fi
     
     if [ -z "$user_email" ]; then
-        read -p "请设置邮箱: " user_email
+        read -e -p "请设置邮箱: " user_email
         if [ -z "$user_email" ]; then
             log_error "邮箱不能为空"
             exit 1
diff -Nuar cqpkg_manager-1.0.1/usr/bin/upload cqpkg_manager-1.0.1-diff/usr/bin/upload
--- cqpkg_manager-1.0.1/usr/bin/upload	2025-06-24 14:41:27.340311333 +0800
+++ cqpkg_manager-1.0.1-diff/usr/bin/upload	2025-06-24 14:43:00.586427341 +0800
@@ -164,7 +164,7 @@
     log_info "开始执行代码上传流程"
     
     # 2. 交互式确认代码来源
-    read -p "是否从 ~/rpmbuild 目录上传代码？(y/yes/其他跳过): " choice
+    read -e -p "是否从 ~/rpmbuild 目录上传代码？(y/yes/其他跳过): " choice
     choice=$(echo "$choice" | tr '[:upper:]' '[:lower:]')
     
     if [[ "$choice" =~ ^(y|yes)$ ]]; then
