diff --git a/lib/utils.sh b/lib/utils.sh
index 7722e98..e82bc93 100644
--- a/lib/utils.sh
+++ b/lib/utils.sh
@@ -100,6 +100,32 @@ parse_args() {
     fi
 }
 
+gitlab_authentication() {
+    if [[ -z "$HOME/.git-credentials" ]]; then
+        touch "$HOME/.git-credentials"
+    fi
+    if grep -q "192.168.10.152" "$HOME/.git-credentials"; then
+        log_warn "未进行身份验证，请验证"
+        gitlab_user=$(read -e -p "请输入gitlab用户名: " reply)
+        gitlab_password=$(read -e -p "请输入gitlab密码: " reply)
+        echo "https://$gitlab_user:$gitlab_password@192.168.10.152" >> "$HOME/.git-credentials"
+    fi
+    git config --global credential.helper store
+}
+
+gitee_authentication() {
+    if [[ -z "$HOME/.git-credentials" ]]; then
+        touch "$HOME/.git-credentials"
+    fi
+    if grep -q "gitee" "$HOME/.git-credentials"; then
+        log_warn "未进行身份验证，请验证"
+        gitee_user=$(read -e -p "请输入gitee用户名: " reply)
+        gitee_password=$(read -e -p "请输入gitee密码: " reply)
+        echo "https://$gitee_user:$gitee_password@192.168.10.152" >> "$HOME/.git-credentials"
+    fi
+    git config --global credential.helper store
+}
+
 # 从%files行提取包名
 extract_pkg() {
     local files_str="$1"
diff --git a/modules/clone.sh b/modules/clone.sh
index a4d6fc1..cfd7a15 100644
--- a/modules/clone.sh
+++ b/modules/clone.sh
@@ -172,6 +172,7 @@ module_clone() {
     # 根据仓库类型克隆
     case "$repo_type" in
         a)
+            gitee_authentication
             local upstream_url="https://gitee.com/src-openeuler/${PKG}.git"
             if ! git clone -b "$branch" "$upstream_url"; then
                 log_error "${PKG}仓库克隆失败！"
@@ -180,6 +181,7 @@ module_clone() {
             log_success "${PKG}克隆成功！"
             ;;
         b)
+            gitee_authentication
             local upstream_url="https://gitee.com/src-anolis-os/${PKG}.git"
             if ! git clone -b "$branch" "$upstream_url"; then
                 log_error "${PKG}克隆失败！"
@@ -188,6 +190,7 @@ module_clone() {
             log_success "${PKG}克隆成功！"
             ;;
         *)
+            gitlab_authentication
             clone_cq_repo
             log_success "${PKG}克隆成功！"
             
