diff -Nuar cqpkg-2.0.1/lib/utils.sh cqpkg-2.0.1-diff/lib/utils.sh
--- cqpkg-2.0.1/lib/utils.sh	2025-09-04 10:22:25.241847250 +0800
+++ cqpkg-2.0.1-diff/lib/utils.sh	2025-09-04 10:38:43.186636565 +0800
@@ -108,7 +108,7 @@
         log_warn "未进行身份验证，请验证"
         gitlab_user=$(read -e -p "请输入gitlab用户名: " && [[ -n "$REPLY" ]] && echo "$REPLY")
         gitlab_password=$(read -e -p "请输入gitlab密码: " && [[ -n "$REPLY" ]] && echo "$REPLY")
-        echo "https://$gitlab_user:$gitlab_password@192.168.10.152" >> "$HOME/.git-credentials"
+        echo "http://$gitlab_user:$gitlab_password@192.168.10.152" >> "$HOME/.git-credentials"
     fi
     git config --global credential.helper store
 }
@@ -119,8 +119,8 @@
     fi
     if ! grep -q "gitee" "$HOME/.git-credentials"; then
         log_warn "未进行身份验证，请验证"
-        gitee_user=$(read -e -p "请输入gitee用户名: " reply)
-        gitee_password=$(read -e -p "请输入gitee密码: " reply)
+        gitee_user=$(read -e -p "请输入gitee用户名: " && [[ -n "$REPLY" ]] && echo "$REPLY")
+        gitee_password=$(read -e -p "请输入gitee密码: " && [[ -n "$REPLY" ]] && echo "$REPLY")
         echo "https://$gitee_user:$gitee_password@192.168.10.152" >> "$HOME/.git-credentials"
     fi
     git config --global credential.helper store
