diff -Nuar cqpkg-2.0.0/modules/compile.sh cqpkg-2.0.0-diff/modules/compile.sh
--- cqpkg-2.0.0/modules/compile.sh	2025-07-09 10:58:34.619501521 +0800
+++ cqpkg-2.0.0-diff/modules/compile.sh	2025-07-09 10:59:00.921748202 +0800
@@ -5,10 +5,10 @@
     if [[ "$build_type" == *"$type"* ]]; then
         rpmbuild -$type -D "_topdir $PKG_PATH" "$SPEC_FILE"
         if [[ $? -ne 0 ]]; then
-            log_error "错误：执行失败！"
+            log_error "错误：$type 执行失败！"
             handle_interrupt
         fi
-        log_success "执行成功！"
+        log_success "$type 执行成功！"
     fi
 }
 # 编译
@@ -27,17 +27,19 @@
     module_rpmbuild "$build_type" "bp"
     module_rpmbuild "$build_type" "bs"
 
-    log_info "\n构建结果："
-    for arch in noarch x86_64; do
-        local arch_dir="$RPM_DIR/$arch"
-        echo -e "${BLUE}== $arch ==${NC}"
-        if [ "$(ls "$arch_dir" 2>/dev/null)" ]; then
-            for rpm_file_path in "$arch_dir"/*; do
-                echo "$rpm_file_path"
-            done
-        else
-            echo "无"
-        fi
-        echo
-    done
+    if [[ "$build_type" == *"ba"* ]]; then
+        log_info "\n构建结果："
+        for arch in noarch x86_64; do
+            local arch_dir="$RPM_DIR/$arch"
+            echo -e "${BLUE}== $arch ==${NC}"
+            if [ "$(ls "$arch_dir" 2>/dev/null)" ]; then
+                for rpm_file_path in "$arch_dir"/*; do
+                    echo "$rpm_file_path"
+                done
+            else
+                echo "无"
+            fi
+            echo
+        done
+    fi
 }
