diff -Nuar cqpkg-2.0.0/modules/clone.sh cqpkg-2.0.0-diff/modules/clone.sh
--- cqpkg-2.0.0/modules/clone.sh	2025-07-09 16:30:01.120191807 +0800
+++ cqpkg-2.0.0-diff/modules/clone.sh	2025-07-09 16:33:32.730763962 +0800
@@ -38,7 +38,7 @@
         fi
     fi
 
-    log_info "整理文件完成"
+    log_success "文件移动完成！"
     return 0
 }
 
@@ -47,8 +47,23 @@
 generate_metadata() {
     cd "$WORK_PATH"
     local SPEC_FILE=$(ls $PKG_PATH/SPECS/*.spec)
-    local url=$(awk '/^(URL|Url):/ {print $2}' "$SPEC_FILE" | head -n 1)
-    local license=$(awk '/^License:/ {print $2}' "$SPEC_FILE" | head -n 1)
+    local NEW_SPEC_FILE="$SPEC_FILE-new"
+    local use_file=""
+    log_action "正在生成元数据文件 SOURCEINFO.yaml ..."
+    if [[ ! -f "$SPEC_FILE" ]]; then
+        log_error "未找到spec文件！"
+        handle_interrupt
+    fi
+    if rpmspec -P "$SPEC_FILE" > "$NEW_SPEC_FILE"; then
+        log_success "宏定义展开成功，生成新spec文件: $NEW_SPEC_FILE"
+        use_file="$NEW_SPEC_FILE"
+    else
+        log_warn "宏定义展开失败，将直接使用原spec文件: $SPEC_FILE"
+        use_file="$SPEC_FILE"
+    fi
+    local url=$(grep -E '^(URL|Url):' "$use_file" | head -n 1 | sed -E 's/^(URL|Url):\s*//')
+    local license=$(grep '^License:' "$use_file" | head -n 1 | sed 's/^License:\s*//')
+    rm -f "$NEW_SPEC_FILE" && log_info "删除新生成的spec文件$NEW_SPEC_FILE"
     # 确定上游仓库URL
     local upstream_url
     if [[ "$repo_type" == "a" ]]; then
@@ -71,7 +86,6 @@
     log_success "元数据文件生成完成 $(pwd)/SOURCEINFO.yaml"
     log_info "查看元数据文件内容: "
     cat $(pwd)/SOURCEINFO.yaml
-    log_info "请根据需要判断是否修改元数据文件内容"
     return 0
 }
 
@@ -208,11 +222,9 @@
     fi
     # 移动文件
     mv_files
-    log_success "文件移动完成！"
 
     # 生成元数据
     generate_metadata
-    log_success "元数据生成成功！"
     
     log_info "操作完成！RPM构建环境已配置到 $WORK_PATH"
 }   
